### PR TITLE
Sophie_B_Airport_Challenge

### PR DESCRIPTION
**Pull request by sofyloafy**

### CODE REVIEW

**Comment by sofyloafy** [comment]
I attempted to cover all user stories, but haven't been able to tie them all together in a coherent programme. 
I would also like feedback on how to make my tests more airtight. I tried to start testing for whether a plane was already in flight, but didn't get too far.

File: spec/airport_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      expect(subject.capacity).to eq Airport::DEFAULT_CAPACITY
+    end
+
+    it "should raise an error if airport is full" do
+      subject.capacity.times { subject.land(plane) }
+      expect { subject.land(plane) }.to raise_error "Airport at capacity"
+    end
+
+    it 'should land a plane' do
+      expect(subject).to respond_to(:land).with(1).argument
```
Comment (RaeRachael) [inline spec/airport_spec.rb:20]: this test is covered in later tests - no longer need?

File: spec/plane_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    airport.land(subject)
+    expect(subject.flying?).to eq false
+  end
+
+  it "should tell a plane to land at an airport" do
+    expect(subject).to respond_to(:land)
+    
+  end
+
+  it "should prevent landing if weather is stormy" do
```
Comment (RaeRachael) [inline spec/plane_spec.rb:21]: how are you forcing the weather to be stormy here?

File: spec/weather_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  describe "#weather" do
+    it { is_expected.to respond_to :current_weather }
+
+    it "have weather conditions" do
+      expect(subject.weather).to eq ["sunny", "sunny", "stormy"]
+    end
+
+    it "gives current weather condition" do
+      allow(subject).to receive(:rand).and_return("stormy")
```
Comment (RaeRachael) [inline spec/weather_spec.rb:14]: does this actually test the code? or is it testing the stubbing?

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @hangar.include?(plane)  
+  end
+
+  def land(plane)
+    fail "Airport at capacity" if full?
+
+    @hangar.push(plane) 
+  end
+
+  def full?
```
Comment (RaeRachael) [inline lib/airport.rb:30]: does this get called from outside of the class? if not maybe make it a private method?

File: lib/plane.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+require 'weather'
+require 'airport'
+
+class Plane
+  attr_accessor :plane
```
Comment (RaeRachael) [inline lib/plane.rb:5]: this choice of name could be confusing, possibly change to :name ?

File: lib/weather.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class Weather
+  def weather
```
Comment (RaeRachael) [inline lib/weather.rb:2]: possibly could be a private method?

File: spec/plane_spec.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
+require 'plane'
+require 'airport'
+
+describe Plane do
+  let(:airport) { Airport.new }
+  let(:plane) { double :plane }
```
Comment (RaeRachael) [inline spec/plane_spec.rb:6]: shouldn't be doubling plane in the plane_spec

File: lib/airport.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    @hangar.delete(plane)
+    true
+  end
+
+  def location(plane)
+    @hangar.include?(plane)  
+  end
+
+  def land(plane)
+    fail "Airport at capacity" if full?
```
Comment (RaeRachael) [inline lib/airport.rb:25]: could possibly be a separate private method

File: spec/plane_spec.rb (hunk 2 above)
Comment (RaeRachael) [reply spec/plane_spec.rb:21]: allow(Weather).to_recieve(:current_weather).and_return('stormy')

**Pull request closed without merging**